File: rust/src/models/recurring_fee.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::billing_type_enum::BillingTypeEnum;

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct RecurringFee {
    pub billing_type: BillingTypeEnum,

    pub quantity: i32,

    pub rate: rust_decimal::Decimal,
}

impl RecurringFee {
    pub fn new(billing_type: BillingTypeEnum, quantity: i32, rate: rust_decimal::Decimal) -> Self {
        Self {
            billing_type,
            quantity,
            rate,
        }
    }
}

File: typescript/src/models/recurringFee.ts
// this file is @generated
import { type BillingTypeEnum, BillingTypeEnumSerializer } from "./billingTypeEnum";

export interface RecurringFee {
  billingType: BillingTypeEnum;

  quantity: number;

  rate: string;
}

export const RecurringFeeSerializer = {
  _fromJsonObject(object: any): RecurringFee {
    return {
      billingType: BillingTypeEnumSerializer._fromJsonObject(object["billing_type"]),
      quantity: object["quantity"],
      rate: object["rate"],
    };
  },

  _toJsonObject(self: RecurringFee): any {
    return {
      billing_type: BillingTypeEnumSerializer._toJsonObject(self.billingType),
      quantity: self.quantity,
      rate: self.rate,
    };
  },
};
